processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -64,6 +68,40 @@ public String apply(AggregatorFactory input)
         );
         combinedAggNames.add(postAgg.getName());
       }
+
+      final Collection<String> aggAndPostAgg = Sets.intersection(
+          Sets.<String>newHashSet(
```

Comment:
Yes, I had originally taken this as a separate group of code to separate functionality, but it actually doesn't look too bad when I put them in the same logic area.